index.html

```diff
@@ -88,49 +94,241 @@
+    svg.append("g")         
+        .attr("class", "grid")
+        .attr("transform", "translate(0," + height + ")")
+        .call(make_x_axis()
+            .tickSize(-height, 0, 0)
+            .tickFormat("")
+        )
+    svg.append("g")         
+        .attr("class", "grid")
+        .call(make_y_axis()
+            .tickSize(-width, 0, 0)
+            .tickFormat("")
+        )
   }
-
+  
   function update() {
-    // TRANSFORM
-    txData = data.slice();
 
+    // TRANSFORM: not using original data 
+    txData = data.slice();
     if (options.yearRange !== 'all') {
       var yearRange = options.yearRange.split('-');
       txData = txData.filter(function (d) {
         return d.year >= yearRange[0] && d.year <= yearRange[1];
       });
     }
-
+    
+  
     // SCATTERPLOT
-
+    var points = svg.selectAll('.point')
+      .data(txData);
+    points.enter().append('circle')
+      .attr('class', 'point')
+      .attr('r', 5);
+    points
+      .attr('cx', function (d) { return x(d.unemployment); })
+      .attr('cy', function (d) { return y(d.employment); });
+    points.exit().remove();
+    
+    
     // HISTOGRAMS
+    var hist = d3.layout.histogram()
+      .value(function (d) {return d.unemployment; })
+      .bins (x.ticks (binCount))
+      (txData);
+
+    hy = d3.scale.linear()
+      .domain([0, d3.max(hist, function (bin) { return bin.length; })])
+      .range ([0, height]); 
+    
+    var barWidth = x(hist[1].x) - x(hist[0].x) + 1 - 50;
+  
+    var bars = svg.selectAll('.bar')
+      .data(hist); 
+
+    //ENTER 
+    bars.enter().append('rect')
+      .attr('class', 'bar')
+      .attr('x', function (d) { return x( d.x ) + 30;  })
+      .attr('width' , barWidth); 
+       
+    // ALL BARS IN HIST 1 
+    
+    bars
+      .attr('y', function (bin) {return height - hy( bin.length ) * 0.1; })
+      .attr('height', function (bin) { return hy(bin.length) * 0.1; })
+     
+
+    // HISTOGRAM2
+    // hist function in d3 
+    // each bin has x, y and dx -- x is the lower bound (first value in the domain of the bin) 
+    // dx width (constant) (total domain divided by the bincount)
+    // x+dx (upper bound of the bin)
+
+    var hist2 = d3.layout.histogram()
+      .value(function (d) {return d.employment; })
+      .bins (y.ticks (binCounty))  
+      (txData);
+
+    x = d3.scale.linear()
+      .domain([0, d3.max(hist2, function (bin) { return bin.length; })])
+      .range ([0, width]);
+    
+    //console.log( hist2[0].x + "," + ( hist2[0].x + hist2[0].dx ) + " : " + hist2[0].y );
+    //console.log( y(hist2[0].y) );
+
+    //console.log( hist2[1].x + "," + ( hist2[1].x + hist2[1].dx ) + " : " + hist2[1].y );
+    //console.log( y(hist2[1].y) );
+    // the function does work 
+
+    var barHeight = y(hist2[0].x) - y(hist2[1].x) - 10;
+    //y - is setting the min and the max of the bin the it passes to each rect so the x, and y coordinates are set. 
+    
+
+    var bars2 = svg.selectAll('.bar')
```

Comment:
So we're saving this selection in a different variable, but we're still selecting anything with the class of `.bar`, which means that the bars for the x axis histogram we created above are going to be replaced by the bars for the y axis histogram. If we gave them each different classes, this would work.